pkg/controller/secretgenerator/secret.go

```diff
@@ -85,3 +101,60 @@ func newSecret(annotations map[string]string) (*Secret, error) {
 
 	return &secret, nil
 }
+
+func createOAuthClient(secret *v1.Secret, cli client.Client) error {
+	secretAnnotations := secret.GetAnnotations()
+	// Return if annotation is missing
+	if len(secretAnnotations) == 0 {
+		return nil
+	}
+	// Return without error if oauth client annotation not found
+	routeName, found := secretAnnotations[SECRET_OAUTH_CLIENT_ANNOTATION]
```

Comment:
It could happen but it's highly unlikely, and as long as we log the error it's fine. We can always force the reconciliation by removing the initial secret an letting the kfctl controller to get it created again.